adapters/export/otlp/metrics.go

```diff
@@ -0,0 +1,169 @@
+package otlp
+
+import (
+	"context"
+	"fmt"
+	"github.com/kubaceg/sofar_g3_lsw3_logger_reader/adapters/devices/sofar"
+	grpc "go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc"
+	http "go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetrichttp"
+	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/metric/global"
+	"go.opentelemetry.io/otel/metric/instrument"
+	sdk "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/metric/metricdata"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.18.0"
+	"log"
+)
+
+const (
+	appName = "sofar.logger"
+)
+
+type Config struct {
+	Http struct {
+		Url string `yaml:"url"`
+	} `yaml:"http"`
+	Grpc struct {
+		Url string `yaml:"url"`
+	} `yaml:"grpc"`
+}
+
+type Service struct {
+	m         metric.Meter
+	reader    sdk.Reader
+	exporters []sdk.Exporter
+}
+
+func New(c *Config) (*Service, error) {
+	reader := sdk.NewManualReader()
+	mp := sdk.NewMeterProvider(
+		sdk.WithReader(reader),
+		sdk.WithResource(newResource()),
+	)
+
+	global.SetMeterProvider(mp)
+	m := global.Meter(appName)
+
+	exporters := make([]sdk.Exporter, 0)
+	if url := c.Grpc.Url; url != "" {
+		e, err := grpc.New(context.Background(), grpc.WithEndpoint(url), grpc.WithInsecure())
+		if err != nil {
+			return nil, err
+		}
+		exporters = append(exporters, e)
+	}
+
+	if url := c.Http.Url; url != "" {
+		e, err := http.New(context.Background(), http.WithEndpoint(url), http.WithInsecure())
+		if err != nil {
+			return nil, err
+		}
+		exporters = append(exporters, e)
+	}
+
+	s := Service{
+		m:         m,
+		reader:    reader,
+		exporters: exporters,
+	}
+
+	err := s.initGauges()
+	if err != nil {
+		return nil, err
+	}
+	return &s, nil
+}
+
+// initGauges creates Int64 gauges for all reply fields that will be read
+func (s *Service) initGauges() error {
+	for _, rr := range sofar.AllRegisterRanges {
+		for _, f := range rr.ReplyFields {
+
+			if f.Name == "" || f.ValueType == "" {
```

Comment:
This could be filtered out on sofar package side if you implement my proposition from lsw.go file comment